### PR TITLE
Fix product creation by admin api

### PR DESCRIPTION
**Pull request by rust-le**
| Q               | A
|-----------------|-----
| Branch?         | 2.1
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | fixes 
| License         | MIT

## Summary by CodeRabbit

- New Features
  - None
- Bug Fixes
  - API product creation now accepts product attributes regardless of whether the attribute reference or its value appears first.
  - Added guards to skip processing when required attribute or value fields are missing, preventing spurious validation errors.
- Tests
  - Added test and fixture verifying product creation succeeds when attribute fields are provided in varying orders.

### CODE REVIEW

File: tests/Api/Admin/ProductsTest.php [hunk 1]
```diff
@@ -147,6 +147,64 @@ public function it_creates_a_product(): void
         );
     }
 
+    #[Test]
+    public function it_creates_a_product_with_value_before_attribute(): void
```
Comment (Rafikooo) [inline tests/Api/Admin/ProductsTest.php:151]: Instead of adding another contract test that duplicates the verification provided by the `it_creates_a_product` test, add a unit test for `ProductAttributeValueDenormalizer` to cover the reorder logic.

File: src/Sylius/Bundle/ApiBundle/Serializer/Denormalizer/ProductAttributeValueDenormalizer.php [hunk 2]
```diff
@@ -77,9 +82,40 @@ private function denormalizeValue(array $data): array
...
+        $result['attribute'] = $data['attribute'];
+
+        foreach ($data as $key => $value) {
+            if ($key !== 'attribute') {
+                $result[$key] = $value;
+            }
+        }
+
+        return $result;
+    }
```
Comment (Rafikooo) [inline src/Sylius/Bundle/ApiBundle/Serializer/Denormalizer/ProductAttributeValueDenormalizer.php:110]:
```suggestion
    /**
     * Reorders the data array to ensure 'attribute' key comes first,
     * preventing setValue() from being called before setAttribute().
     *
     * @see AttributeValue::setValue()
     *
     * @param array<string, mixed> $data
     *
     * @return array<string, mixed>
     */
    private function reorderAttributeFirst(array $data): array
    {
        $attribute = $data['attribute'];
        unset($data['attribute']);

        return ['attribute' => $attribute, ...$data];
    }
```

File: src/Sylius/Bundle/ApiBundle/Serializer/Denormalizer/ProductAttributeValueDenormalizer.php [hunk 3]
```diff
@@ -51,6 +51,7 @@ public function denormalize(mixed $data, string $type, ?string $format = null, a
 
         $this->validateValue($data);
         $data = $this->denormalizeValue($data);
+        $data = $this->ensureAttributeFirst($data);
 
         return $this->denormalizer->denormalize($data, $type, $format, $context);
```
Comment (Rafikooo) [inline src/Sylius/Bundle/ApiBundle/Serializer/Denormalizer/ProductAttributeValueDenormalizer.php:56]:
```suggestion
        return $this->denormalizer->denormalize($this->reorderAttributeFirst($data), $type, $format, $context);
```

File: src/Sylius/Bundle/ApiBundle/Serializer/Denormalizer/ProductAttributeValueDenormalizer.php [hunk 4]
```diff
@@ -51,6 +51,7 @@ public function denormalize(mixed $data, string $type, ?string $format = null, a
 
         $this->validateValue($data);
         $data = $this->denormalizeValue($data);
+        $data = $this->ensureAttributeFirst($data);
```
Comment (Rafikooo) [inline src/Sylius/Bundle/ApiBundle/Serializer/Denormalizer/ProductAttributeValueDenormalizer.php:54]: Can be inlined.
```suggestion
```

File: src/Sylius/Bundle/ApiBundle/Serializer/Denormalizer/ProductAttributeValueDenormalizer.php [hunk 5]
```diff
@@ -67,6 +68,10 @@ public function getSupportedTypes(?string $format): array
      */
     private function denormalizeValue(array $data): array
```
Comment (Rafikooo) [inline src/Sylius/Bundle/ApiBundle/Serializer/Denormalizer/ProductAttributeValueDenormalizer.php:69]: And then no move checks or function calls needed.
```suggestion
    private function denormalizeValue(array $data, ProductAttributeInterface $attribute): array
```

File: src/Sylius/Bundle/ApiBundle/Serializer/Denormalizer/ProductAttributeValueDenormalizer.php [hunk 6]
```diff
@@ -67,6 +68,10 @@ public function getSupportedTypes(?string $format): array
...
     {
+        if (!isset($data['attribute']) || !isset($data['value'])) {
+            return $data;
+        }
+
         /** @var ProductAttributeInterface $attribute */
         $attribute = $this->iriConverter->getResourceFromIri($data['attribute']);
```
Comment (Rafikooo) [inline src/Sylius/Bundle/ApiBundle/Serializer/Denormalizer/ProductAttributeValueDenormalizer.php:76]:
```suggestion
```

File: src/Sylius/Bundle/ApiBundle/Serializer/Denormalizer/ProductAttributeValueDenormalizer.php [hunk 7]
```diff
@@ -51,6 +51,7 @@ public function denormalize(mixed $data, string $type, ?string $format = null, a
```
Comment (Rafikooo) [inline src/Sylius/Bundle/ApiBundle/Serializer/Denormalizer/ProductAttributeValueDenormalizer.php:51]: Skip custom processing if attribute is missing.
```suggestion
        if (!isset($data['attribute'])) {
            return $this->denormalizer->denormalize($data, $type, $format, $context);
        }
```

**Comment by Rafikooo** [comment]
I'm closing this PR in favor of: https://github.com/Sylius/Sylius/pull/18484
